### PR TITLE
chore: replace usage of public class field assignments

### PR DESCRIPTION
**Pull request by boneskull**
Basically, public class fields don't work the way you'd expect, so I'd like to avoid assignment to them whenever possible.  Compounding the situation is the fact that `tsc` (and Babel, fwiw) both transpile in a way that deviates slightly from the official specification.

Given the following example:

```js
class A {
  value = 'a';
  constructor() {
    console.log(this.value);
  }
}

class B extends A {
  value = 'b';
}
```

What is logged by `new B()`?  If you guessed `b`, you'd be both reasonable and wrong.  What's happening is that the super class does not know anything about public class fields in subclasses.  If `value` was instead a method or getter, this _would_ work as you'd expect.

Probably better to avoid them whenever possible.  It is perfectly safe to _declare_ them for type inference & the like, but assignment should be avoided until it's absolutely the right thing to do.

Note that `tsc` has a flag to change the transpilation behavior to act in-line with the spec, but it does not change anything in the above regard, AFAICT.

### CODE REVIEW

**Comment by boneskull** [comment]
To be clear: `static` fields, private `static` fields, and private class fields do not suffer from this problem

File: packages/base-driver/lib/basedriver/core.js [hunk 1]
```diff
@@ -56,39 +57,43 @@ class DriverCore {
...
-  managedDrivers = [];
+  managedDrivers;
 
   /** @type {NodeJS.Timeout?} */
-  noCommandTimer = null;
+  noCommandTimer;
 
   /** @type {EventHistory} */
-  _eventHistory = {commands: []};
+  _eventHistory;
```
Comment (mykola-mokhnach) [inline packages/base-driver/lib/basedriver/core.js:92]: should it be also changed to `#eventHistory` ?

File: packages/fake-driver/lib/driver.js [hunk 2]
```diff
@@ -15,20 +15,45 @@ const FAKE_DRIVER_CONSTRAINTS = /** @type {const} */ ({
...
 
   /** @type {string} */
   curContext;
 
-  appModel = new FakeApp();
+  /** @type {FakeApp} */
+  appModel;
+
+  /** @type {boolean} */
+  _proxyActive;
```
Comment (mykola-mokhnach) [inline packages/fake-driver/lib/driver.js:41]: same as above

File: packages/universal-xml-plugin/lib/plugin.js [hunk 3]
```diff
@@ -1,20 +1,13 @@
...
 
-import BasePlugin from 'appium/plugin';
+import {BasePlugin} from 'appium/plugin';
 import {errors} from 'appium/driver';
 import {transformSourceXml} from './source';
 import {transformQuery} from './xpath';
 import log from './logger';
 
 export default class UniversalXMLPlugin extends BasePlugin {
-  commands = [
```
Comment (mykola-mokhnach) [inline packages/universal-xml-plugin/lib/plugin.js:10]: how this assignment should work now?

Comment (boneskull) [reply packages/universal-xml-plugin/lib/plugin.js:10]: this code was unused afaict.

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18208** <a href="https://app.graphite.dev/github/pr/appium/appium/18208" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18208?utm_source=stack-comment).

File: packages/fake-driver/lib/driver.js [hunk 4]
```diff
@@ -205,82 +232,354 @@ export class FakeDriver extends BaseDriver {
   /*********
    * ALERT *
    *********/
-  assertNoAlert = alert.assertNoAlert;
-  assertAlert = alert.assertAlert;
-  getAlertText = alert.getAlertText;
-  setAlertText = alert.setAlertText;
-  postAcceptAlert = alert.postAcceptAlert;
-  postDismissAlert = alert.postDismissAlert;
+  /**
```
Comment (jlipps) [inline packages/fake-driver/lib/driver.js:235]: ugh i hate that one line is now 6. it makes it much less desirable to use this file-splitting pattern

File: packages/universal-xml-plugin/lib/plugin.js (hunk 3 above)
Comment (jlipps) [reply packages/universal-xml-plugin/lib/plugin.js:10]: yeah i think it came from an old version of our base plugin logic which required registration of overrides.

**Review by jlipps — APPROVED** [review]

File: packages/fake-driver/lib/driver.js (hunk 4 above)
Comment (boneskull) [reply packages/fake-driver/lib/driver.js:235]: I kind of hate it too.  there's another way, which is to just fake it.  let me see if I can do that.

**Comment by boneskull** [comment]
I've changed this to exclude changes to `fake-driver`, since they were more involved; see #18237

**Comment by boneskull** [comment]
labeled "enhancement" since `@appium/types` gets a new type

**Review by mykola-mokhnach — APPROVED** [review]

**Review by jlipps — APPROVED** [review]

**Pull request merged**